Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.ts

```diff
@@ -0,0 +1,353 @@
+          getDropDownSearchString: (searchString: string) =>
+          {
+            return searchStringValidationInfo.IgnoreSearchStringForDropDownUponValidInput ? GetDropDownSearchStringForValidInput(searchString) : searchString;
+          }
+        },
+        max: completionSetCount, // Set query limit
+
+        dropDownDisplayDelay: dropDownDisplayDelay,
+        dropDownRefreshDelay: dropDownRefreshDelay,
+        selectionUpdateDelay: selectionUpdateDelay,
+
+        noDataFoundMessage: resources.NoDataFoundMessage,
+        autoFill: true,
+        matchContains: true,
+        combobox: combobox,
+        selectListID: this._selectListID,
+        informationPopUpID: this._informationPopUpID,
+        dropDownButtonID: button.attr('id'),
+        inputAreaClass: 'content',
+        // this can be set to true/removed once the problem is fixed that an empty textbox still selects the first element, making it impossible to clear the selection
+        selectFirst: function (inputValue: string, searchTerm: string)
+        {
+          return inputValue.length > 0;
+        },
+        dataType: 'json',
+        parse: function (data: any)
+        {
+          return $.map(data, function (row)
+          {
+            row.IsAnnotation = row.UniqueIdentifier === nullValueString;
+
+            return {
+              data: row,
+              value: row.UniqueIdentifier,
+              result: row.IsAnnotation ? '' : row.DisplayName
+            };
+          });
+        },
+        formatItem: function (item: JQuery) //What we display on input box
+        {
+          var row = $('<li/>');
+
+          if (item.IconUrl != '')
+          {
+            var img = $('<img/>');
+            img.attr({ src: item.IconUrl, alt: '', 'aria-hidden': 'true' });
+            row.append($('<div/>').append(img));
+          }
+
+          var displayName = $('<span/>');
+          displayName.text(item.DisplayName);
+          row.append($('<div/>').append(displayName));
+
+          return {
+            html: row.html(),
+            class: null,
+            isAnnotation: item.IsAnnotation
+          };
+        },
+        formatMatch: function (item: BocAutoCompleteReferenceValue_Item) //The value used by the cache
+        {
+          return item.DisplayName;
+        },
+        handleRequestError: function (err: Sys.Net.WebServiceError)
+        {
+          SetError(resources.LoadDataFailedErrorMessage);
+        },
+        clearRequestError: function ()
+        {
+          ClearError();
+        }
+      }
+    ).invalidateResult(() =>
+    {
+      if (this._isInvalidated)
+        return;
+
+        this._isInvalidated = true;
+
+      hiddenField.val(nullValueString);
+      UpdateCommand(nullValueString);
+      //Do not fire change-event
+    }).updateResult((e: any, item: BocAutoCompleteReferenceValue_Item, out: any) => //TODO what is this
```

Comment:
The type of `out` is `{ Value: Nullable<BocAutoCompleteReferenceValue_Item> }` but we should create a separate type for it. Maybe `BocAutoCompleteReferenceValue_UpdateResult` and use it here as type. And `e` is of type `JQueryEventObject`.